pages/settings/wallets/nwc.js

```diff
@@ -0,0 +1,71 @@
+import { getGetServerSideProps } from '../../../api/ssrApollo'
+import { Form, ClientInput } from '../../../components/form'
+import { CenterLayout } from '../../../components/layout'
+import { WalletButtonBar, WalletCard } from '../../../components/wallet-card'
+import { nwcSchema } from '../../../lib/validate'
+import { useToast } from '../../../components/toast'
+import { useRouter } from 'next/router'
+import { useNWC } from '../../../components/webln/nwc'
+
+export const getServerSideProps = getGetServerSideProps({ authRequired: true })
+
+export default function NWC () {
+  const { nwcUrl, saveConfig, clearConfig, enabled } = useNWC()
+  const toaster = useToast()
+  const router = useRouter()
+
+  return (
+    <CenterLayout>
+      <h2 className='pb-2'>nwc</h2>
+      <h6 className='text-muted text-center pb-3'>use Nostr Wallet Connect for zapping</h6>
+      <Form
+        initial={{
+          nwcUrl: nwcUrl || ''
+        }}
+        schema={nwcSchema}
+        onSubmit={async (values) => {
+          try {
+            await saveConfig(values)
+            toaster.success('saved settings')
+            router.push('/settings/wallets')
+          } catch (err) {
+            console.error(err)
+            toaster.danger('failed to attach: ' + err.message || err.toString?.())
+          }
+        }}
+      >
+        <ClientInput
+          initialValue={nwcUrl}
+          label='connection'
+          name='nwcUrl'
+          required
+          autoFocus
+        />
+        <WalletButtonBar
+          enabled={enabled} onDelete={async () => {
+            try {
+              await clearConfig()
+              toaster.success('saved settings')
+              router.push('/settings/wallets')
+            } catch (err) {
+              console.error(err)
+              toaster.danger('failed to unattach: ' + err.message || err.toString?.())
+            }
+          }}
+        />
+      </Form>
+    </CenterLayout>
+  )
+}
+
+export function NWCCard () {
+  const { enabled } = useNWC()
+  return (
+    <WalletCard
+      title='nwc'
+      badges={['send only', 'non-custodialish', 'async']}
```

Comment:
Mhh, not sure about using `async` here. Might not be clear what is meant with that. I essentially what to mention that this wallet connection might require confirmation (if no budget is set) and supports budgets (payments within the budget are automatically approved). Some ideas I had: `budgetable` and `requires confirmation`

Reply:
`budgetable` might be the most straightforward. I think we'll need to revisit the UI some in the future so we can fit things like `requires confirmation`.